sqlglot/optimizer/scope.py

```diff
@@ -616,7 +616,7 @@ def _is_derived_table(expression: exp.Subquery) -> bool:
     as it doesn't introduce a new scope. If an alias is present, it shadows all names
     under the Subquery, so that's one exception to this rule.
     """
-    return bool(expression.alias or not isinstance(expression.unnest(), exp.Table))
+    return bool(expression.alias or isinstance(expression.this, exp.Subqueryable))
```

Comment:
is this and the other isinstance check on 127 and 742 the bulk of the change? it seems like everything else is just handling the side effect of that?

Reply:
more or less -- all changes in the scope module were key and the rest were just side effects, yes